File: PasswordGen/Password_Generator v1.pl
# Basic Password Generator in Perl
# This script generates a random password of a specified length using letters, digits, and punctuation.

use strict;
use warnings;

my @all_characters = ('A'..'Z', 'a'..'z', '0'..'9', split(//, '!\"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~'));

print "Enter the length of the password: ";
chomp(my $length = <STDIN>);

my $password = '';
$password .= $all_characters[int(rand(@all_characters))] for 1..$length;

print "Your password is: $password\n";

File: PasswordGen/Password_Generator v1.js
// Basic Password Generator in JavaScript
// This script generates a random password of a specified length using letters, digits, and punctuation.

import { createInterface } from "readline";

const rl = createInterface({
  input: process.stdin,
  output: process.stdout
});

const allCharacters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~";

rl.question("Enter the length of the password: ", function(length) {
  let password = '';
  for (let i = 0; i < parseInt(length); i++) {
    password += allCharacters.charAt(Math.floor(Math.random() * allCharacters.length));
  }
  console.log("Your password is:", password);
  rl.close();
});
